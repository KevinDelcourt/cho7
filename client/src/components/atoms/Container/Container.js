import styled from "styled-components"
import { connect } from "react-redux"

const Container = styled.div`
    background-color: ${props => props.bgColor};
    border-radius: ${props => props.borderRadius};
    border: ${props => props.borderSize} solid ${props => props.color};
    font-size: ${props => props.fontSize};
    font-family: ${props => props.font};
    color: ${props => props.color};
    padding: 20px 30px;
    box-shadow: ${props => props.boxShadow || "none"};
    width: ${props => props.width || "100%"};
    margin-left: auto;
    margin-right: auto;
`

const mapStateToProps = state => {
    return {
        bgColor: state.app.theme.colorContainerBg,
        borderRadius: state.app.theme.borderRadius,
        borderSize: state.app.theme.borderSize,
        fontSize: state.app.theme.fontSizeText,
        font: state.app.theme.fontBase,
        color: state.app.theme.colorText
    }
}

export default connect(mapStateToProps)(Container)
