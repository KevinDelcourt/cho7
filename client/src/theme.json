{
    "color": {
        "brown2": "#916D43",
        "brown1": "rgba(145, 109, 67, 0.35)",
        "lightgrey1": "rgba(255, 255, 255, 0.54)",
        "grey1": "#C4C4C4",
        "grey2": "#eaeaea",
        "orange1": "#ef7513",
        "orange2": "#b8590c",
        "orange3": "#af540c",
        "white": "#fff"
    },
    "effect": {
        "shadow": "0px 4px 15px rgba(0, 0, 0, 0.25)"
    },
    "fontFamily": {
        "ruluko": "Ruluko"
    }
}
