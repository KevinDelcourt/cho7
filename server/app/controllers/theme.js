const { queryCallback } = require("../../modules/queries")

const themeController = connection => {
    return {
        getTheme: (req, res) => {
            connection.query("SELECT * FROM theme", (err, rows) =>
                queryCallback(err, res, () => {
                    let theme = {}
                    rows.map(r => {
                        theme[r.style] = r.value
                    })
                    return res.send(theme)
                })
            )
        },

        updateTheme: (req, res) => {
            let query = {
                mysql: "INSERT INTO theme (style,value) VALUES ",
                data: []
            }

            const addValues = (key, value) => {
                query.mysql += "(?,?),"
                query.data.push(key)
                query.data.push(value)
            }

            for (let key in req.body)
                if (req.body[key] != "undefined") addValues(key, req.body[key])

            for (let key in req.files)
                addValues(key.slice(0, -4), req.files[key][0].originalname)

            query.mysql = query.mysql.slice(0, -1)
            connection.query("DELETE FROM theme", (err, rows) => {
                queryCallback(err, res, () => {
                    connection.query(query.mysql, query.data, (err, rows) => {
                        queryCallback(err, res, () => res.send(true))
                    })
                })
            })
        }
    }
}

module.exports = themeController
