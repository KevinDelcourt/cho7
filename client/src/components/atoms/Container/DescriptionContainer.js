import Container from "./Container"
import styled from "styled-components"
import { connect } from "react-redux"

const DescriptionContainer = styled(Container)`
    background-color: ${props => props.bgColor};
    border-radius: ${props => props.borderRadius};
    border: ${props => props.borderSize} solid ${props => props.color};
    font-size: ${props => props.fontSize};
    font-family: ${props => props.font};
    color: ${props => props.color};
    width: ${props => props.width || "100%"};
    margin: 10px 0;
    padding: 5px 10px;
    overflow-wrap: break-word;
`

const mapStateToProps = state => {
    return {
        bgColor: state.app.theme.colorDescriptionBg,
        borderRadius: state.app.theme.borderRadius,
        fontSize: state.app.theme.fontSizeText,
        borderSize: state.app.theme.borderSize,
        font: state.app.theme.fontBase,
        color: state.app.theme.colorText
    }
}

export default connect(mapStateToProps)(DescriptionContainer)
