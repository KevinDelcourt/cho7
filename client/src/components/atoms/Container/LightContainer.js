import Container from "./Container"
import styled from "styled-components"
import { connect } from "react-redux"

const LightContainer = styled(Container)`
    background-color: ${props => props.bgColor};
    border-radius: ${props => props.borderRadius};
    border: ${props => props.borderSize} solid ${props => props.color};
    color: ${props => props.color};
    width: auto;
    padding: 5px 1vw;
    display: flex;
    justify-content: space-between;
    margin: 0 2vw;
`

const mapStateToProps = state => {
    return {
        bgColor: state.app.theme.colorMenuBarBg,
        borderRadius: state.app.theme.borderRadius,
        borderSize: state.app.theme.borderSize,
        color: state.app.theme.colorText
    }
}

export default connect(mapStateToProps)(LightContainer)
