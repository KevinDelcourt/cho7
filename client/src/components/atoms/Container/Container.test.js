import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Container from "./Container"
import theme from "../../../theme.json"

test("Container background-color renders correctly", () => {
    const tree = renderer
        .create(<Container bgColor={theme.color.brown1} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("background-color", "rgba(145,109,67,0.35)")
})

test("Container border-radius renders correctly", () => {
    const tree = renderer.create(<Container borderRadius="20px" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("border-radius", "20px")
})

test("Container boxShadow renders correctly", () => {
    const tree = renderer
        .create(<Container boxShadow={theme.effect.shadow} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("box-shadow", "0px 4px 15px rgba(0,0,0,0.25)")
})

test("Container width renders correctly", () => {
    const tree = renderer.create(<Container width="20px" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("width", "20px")
})
