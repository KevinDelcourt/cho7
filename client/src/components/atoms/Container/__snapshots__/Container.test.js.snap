// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container background-color renders correctly 1`] = `
.c0 {
  background-color: rgba(145,109,67,0.35);
  border-radius: 20px;
  padding: 20px 30px;
  box-shadow: none;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
}

<div
  className="c0"
/>
`;

exports[`Container border-radius renders correctly 1`] = `
.c0 {
  background-color: rgba(145,109,67,0.35);
  border-radius: 20px;
  padding: 20px 30px;
  box-shadow: none;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
}

<div
  className="c0"
/>
`;

exports[`Container boxShadow renders correctly 1`] = `
.c0 {
  background-color: rgba(145,109,67,0.35);
  border-radius: 20px;
  padding: 20px 30px;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
}

<div
  className="c0"
/>
`;

exports[`Container width renders correctly 1`] = `
.c0 {
  background-color: rgba(145,109,67,0.35);
  border-radius: 20px;
  padding: 20px 30px;
  box-shadow: none;
  width: 20px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
}

<div
  className="c0"
  width="20px"
/>
`;
